src/components/AudioPlayer/RepeatAudioModal/RepeatAudioModal.tsx

```diff
@@ -0,0 +1,127 @@
+    const keys = generateChapterVersesKeys(chapterId);
+
+    const initialState = keys.map((chapterVersesKey) => ({
+      id: chapterVersesKey,
+      name: chapterVersesKey,
+      value: chapterVersesKey,
+      label: chapterVersesKey,
+    }));
+
+    return initialState;
+  }, [chapterId]);
+
+  const firstVerseKeyInThisChapter = comboboxVerseItems[0].value;
+  const lastVerseKeyInThisChapter = comboboxVerseItems[comboboxVerseItems.length - 1].value;
+
+  const [verseRepetition, setVerseRepetition] = useState({
+    repeatRange: 1,
+    repeatEachVerse: 1,
+    from: defaultSelectedVerse || firstVerseKeyInThisChapter,
+    to: lastVerseKeyInThisChapter,
+  });
+  const [delayMultiplierBetweenVerse, setDelayMultiplierBetweenVerse] = useState(0);
+
+  // reset verseRepetition's `to` and `from`, when chapter changed
+  useEffect(() => {
+    setVerseRepetition((prevVerseRepetition) => ({
+      ...prevVerseRepetition,
+      from: defaultSelectedVerse || firstVerseKeyInThisChapter,
+      to: lastVerseKeyInThisChapter,
+    }));
+  }, [chapterId, firstVerseKeyInThisChapter, lastVerseKeyInThisChapter, defaultSelectedVerse]);
+
+  const onPlayClick = () => {
+    onClose();
+  };
+  const onCancelClick = () => {
+    onClose();
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClickOutside={onClose}>
+      <Modal.Body>
+        <Modal.Header>
+          <Modal.Title>Repeat Settings</Modal.Title>
+          <Modal.Subtitle>Surah {chapterName}</Modal.Subtitle>
+        </Modal.Header>
+        <div>
+          <SelectRepetitionMode
+            defaultRepetitionMode={defaultRepetitionMode}
+            rangeEndVerse={verseRepetition.to}
+            rangeStartVerse={verseRepetition.from}
+            comboboxVerseItems={comboboxVerseItems}
+            onSingleVerseChange={(val) =>
+              setVerseRepetition({ ...verseRepetition, from: val, to: val })
+            }
+            onRangeChange={(val) => setVerseRepetition({ ...verseRepetition, ...val })}
+            verseKey={verseRepetition.from}
+          />
+          <div className={styles.separator}>
+            <Separator />
+          </div>
+          <RepeatSetting
+            label="Repeat each verse"
+            value={verseRepetition.repeatEachVerse}
+            minValue={1}
+            onChange={(val) => setVerseRepetition({ ...verseRepetition, repeatEachVerse: val })}
+            suffix="times"
+          />
+          <RepeatSetting
+            label="Repeat each range"
+            value={verseRepetition.repeatRange}
+            minValue={1}
+            onChange={(val) => setVerseRepetition({ ...verseRepetition, repeatRange: val })}
+            suffix="times"
+          />
+          <RepeatSetting
+            label="Delay between verse"
+            value={delayMultiplierBetweenVerse}
+            minValue={0}
+            onChange={(val) => setDelayMultiplierBetweenVerse(val)}
+            suffix="times"
```

Comment:
This should be seconds, step 1. A dropdown would be better here IMO.

Reply:
i see,
we planned to make the delay relative to the duration of previous ayah. So it will be 1x,2x,3x previous ayah.
maybe the suffix should `x prev ayah` ?

what do you think @naveed-ahmad